src/cache.rs

```diff
@@ -0,0 +1,28 @@
+use jsonrpsee::core::JsonValue;
+use lru::LruCache;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+pub struct Cache {
+    map: Arc<Mutex<LruCache<String, JsonValue>>>,
+}
+
+impl Cache {
+    pub fn new(size: NonZeroUsize) -> Self {
+        Self {
+            map: Arc::new(Mutex::new(LruCache::new(size))),
+        }
+    }
+
+    pub async fn get(&self, key: &String) -> Option<JsonValue> {
+        self.map.lock().await.get(key).cloned()
+    }
+
+    pub fn put(&self, key: String, value: JsonValue) {
+        let map = self.map.clone();
+        tokio::spawn(async move {
+            let _ = map.lock().await.put(key, value);
```

Comment:
don't just ignore error. at least log it

Reply:
No error, ignoring old value